syntax = "proto3";
package Responses;

// Response after client requests a region
message RegionReadResponse {
    // for 4-byte alignment purposes, put bytes and repeated ints first
    bytes compressed_image_data = 1;
    repeated float image_data = 2;
    repeated int32 nan_encodings = 3;
    bool success = 4;
    int32 x = 5;
    int32 y = 6;
    int32 channel = 7;
    int32 width = 8;
    int32 height = 9;
    int32 mip = 10;
    int32 compression = 11;
    int32 num_values = 12;
    Statistics stats = 13;
}

message Statistics {
    float min_val = 1;
    float max_val = 2;
    float mean = 3;
    float nan_counts = 4;
    Histogram hist = 5;
    Percentiles percentiles = 6;
}

message Histogram {
    repeated int32 bins = 1;
    int32 N = 2;
    float bin_width = 3;
    float first_bin_center = 4;
}

message Percentiles {
    repeated float percentiles = 1;
    repeated float values = 2;
}